app/Commands/NewCommand.php

```diff
@@ -0,0 +1,474 @@
+        $this->projecturl = 'http://'.$this->projectname.'.dev';
+
+        $this->getAvailableTools();
+        if (! $this->tools['laravel']) {
+            $this->error('Unable to find laravel installer so I must exit. One day I will use composer here instead of exiting.');
+            exit;
+        }
+
+        $this->setBasePath();
+
+        $this->projectpath = $this->basepath.DIRECTORY_SEPARATOR.$this->projectname;
+
+        if (is_dir($this->projectpath)) {
+            if (! $this->askToAndRemoveProject()) {
+                $this->error('Goodbye!');
+                exit;
+            }
+        }
+
+        if ($this->option('dev')) {
+            $this->warn('The laravel installation will use the latest developmental branch by passing in --dev');
+            $branch = ' --dev';
+        } else {
+            $branch = '';
+        }
+
+        $command = "laravel new {$this->projectname}$branch";
+
+        $this->info("Creating a new project named {$this->projectname}");
+        $this->info("Executing command '$command' in directory {$this->basepath}");
+
+        $process = new Process($command);
+        $process->setWorkingDirectory($this->basepath);
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+
+        // @todo Determine why the above outputs before this point; so the following getOutput() call does nothing
+        $this->line($process->getOutput());
+
+        if ($this->replaceEnvVariables()) {
+            $this->info('I replaced .env variables in your new Laravel application');
+        }
+
+        if ($this->option('auth')) {
+            $this->doAuth();
+        }
+
+        if ($this->option('createdb')) {
+            if (! in_array($this->option('createdb'), $this->dbtypes)) {
+                $this->info("You passed in '--createdb ".$this->option('createdb')."' but I do not understand");
+                $type = $this->anticipate('What type of database would you like to install? Options are '.implode(' or ', $this->dbtypes).'.', $this->dbtypes, $this->dbtypes[0]);
+            } else {
+                $type = $this->option('createdb');
+            }
+
+            if (! in_array($type, $this->dbtypes)) {
+                $this->alert("Now you're being silly. Entering $type, really? Okay, I won't create a database for you.");
+            } else {
+                $this->info("I am creating a new $type database");
+                if ($this->createDatabase($type)) {
+                    $this->info("I am executing 'php artisan migrate:fresh'");
+                    $this->migrateFresh();
+                }
+            }
+        }
+
+        if ($this->option('node')) {
+            $this->doNodeOrYarn();
+        }
+
+        if ($this->option('link')) {
+            if ($this->tools['valet']) {
+                if ($this->doValetLink()) {
+                    $this->openBrowser();
```

Comment:
I actually want the browser to open even if we don't have link setup; I have Valet share set up so I don't have to manually link it every time, but I still want it to open.

Reply:
That makes sense, although the original intent was to only open the browser if `valet` was installed on the system but clearly that's not the implemented logic :) 

Are you okay with only opening the browser if valet is installed? Later this can be a configuration option of some sort as I'm guessing some non-valet users will want the browser opened.

Reply:
I think it definitely makes sense to only run it if Valet is installed 👌

Reply:
Fixed via commit 03c4a23